src/sfCommand.ts

```diff
@@ -138,11 +165,65 @@ export abstract class SfCommand<T> extends Command {
     };
   }
 
+  protected async catch(error: SfCommand.Error): Promise<SfCommand.Error> {
+    process.exitCode = process.exitCode ?? error.exitCode ?? 1;
+    this.log(this.formatError(error));
+    if (this.jsonEnabled()) {
+      CliUx.ux.styledJSON(this.toErrorJson(error));
+    }
+    return error;
+  }
+
+  /**
+   * Format errors and actions for human consumption. Adds 'Error:',
+   * When there are actions, we add 'Try this:' in blue
+   * followed by each action in red on its own line.
+   * If Error.code is present it is output last in parentheses
+   *
+   * @returns {string} Returns decorated messages.
+   */
+  protected formatError(error: SfCommand.Error): string {
+    const colorizedArgs: string[] = [];
+    colorizedArgs.push(`${chalk.bold.red(messages.getMessage('error.prefix'))} ${error.message}`);
+    colorizedArgs.push(...this.formatActions(error.actions || []));
+    if (error.stack && envVars.getString(SfCommand.SF_ENV) === Mode.DEVELOPMENT) {
+      colorizedArgs.push(chalk.red(`\n*** Internal Diagnostic ***\n\n${error.stack}\n******\n`));
+    }
+    if (error.code) {
+      colorizedArgs.push(chalk.bold(`\n(${error.code})`));
+    }
+
+    return colorizedArgs.join('\n');
+  }
+
+  /**
+   * Utility function to format actions lines
+   *
+   * @param actions
+   * @param options
+   * @private
+   */
+  private formatActions(
+    actions: string[],
+    options: { actionColor: typeof chalk } = { actionColor: chalk.red }
+  ): string[] {
+    const colorizedArgs: string[] = [];
+    // Format any actions.
+    if (actions?.length) {
+      colorizedArgs.push(`\n${chalk.blue.bold(messages.getMessage('actions.tryThis'))}\n`);
+      actions.forEach((action) => {
+        colorizedArgs.push(`${options.actionColor(action)}`);
+      });
+    }
+    return colorizedArgs;
+  }
+
   public abstract run(): Promise<T>;
 }
 
 export namespace SfCommand {
-  export type Warning = string | Error;
+  export type Info = SfCommand.Error | string;
```

Comment:
Per my comment [here](https://github.com/forcedotcom/sfdx-core/pull/526#discussion_r813107949) I think I'd prefer to keep the `Info` type as a string.

The only reason `Warning` ever included `Error` was because that's what we get from the base `warn` method in oclif/core: https://github.com/oclif/core/blob/main/src/command.ts#L176

Reply:
@peternhale should this now be a `StructuredMessage`? https://github.com/forcedotcom/sfdx-core/pull/526/files#diff-46bbb95b17ce5a3865ddffffb0fb655706c57eeb0b4a0d5fee7b2a4ed6564723R35

Reply:
@mdonnalley yes, waiting for sfdx-core to be merged so I can make apply the new structure here.